### DISCUSSION TITLE
Question regarding Bubbletea's internals

### DISCUSSION THREAD

**Discussion by leamandeep** [charmbracelet/bubbletea#822] [General]
Hello 👋🏻  Charm's Team! Bubble tea is an awesome product. You can build a TUI app effortlessly in no time. 
My question is how does bubble tea work? I have gone through the readme. But it just scratches the surfaces. Is there any material, docs with an in-depth overview of how bubble tea renders, etc....

Thanks in advance

**Comment by bashbunni** [comment]
Hey @elamandeep I'm not sure how deep you're hoping to go here... If I were to guess, I would start with reviewing the [API](https://pkg.go.dev/github.com/charmbracelet/bubbletea#section-documentation) and clicking on the specific functionality you're curious about and walking through the source code that way. 

I'm not sure if that's the kind of direction you're looking for or if you're asking about timing and design of the application rather than the actual moving parts. Either way, let me know :)
